team-telnyx/telnyx-webrtc-flutter
# feat(WEBRTC-2079): Actions Unit Tests + Static Code Analysis fixes

[WEBRTC-2079 - Actions Unit Tests + Static Code Analysis fixes.](https://telnyx.atlassian.net/browse/WEBRTC-2079)
---


This PR implements a Github Action to run Unit tests on Push, Pull Request and a cron job schedule. The scheduled unit tests act as a sort of BlackBox test for things like socket connections, etc. 

Additionally, we have modified some code as suggested by static code analysis (Formatting, unnecessary null checks, etc.).

Lastly, we have added an example folder which points to the general readme and links to the core folder of the repo which currently serves as our example. Included in all Readmes we have also added a pub dev and unit test badge to show those states. 

## :older_man: :baby: Behaviors
### Before changes
No automated tests or example folder for pub dev

### After changes
- Automated tests
- Example folder linking to core repo folder via readme
- Static code analysis changes
- Readme badges 

## ✋ Manual testing
1. View readmes and use badges (badges)
2. Run app as usual to test functionality (static code analysis changes)
3. Make sure tests are running on this PR (Automated tests)
